packages/spatial/src/renderer/components/HighlightComponent.ts

```diff
@@ -26,23 +26,31 @@ Ethereal Engine. All Rights Reserved.
 import { defineQuery, defineSystem, Engine } from '@etherealengine/ecs'
 import { defineComponent, getComponent, hasComponent } from '@etherealengine/ecs/src/ComponentFunctions'
 
+import { Object3D } from 'three'
+import { traverseEntityNode } from '../../transform/components/EntityTree'
 import { RendererComponent, WebGLRendererSystem } from '../WebGLRendererSystem'
 import { GroupComponent } from './GroupComponent'
 import { MeshComponent } from './MeshComponent'
 import { VisibleComponent } from './VisibleComponent'
 
 export const HighlightComponent = defineComponent({ name: 'HighlightComponent' })
 
-const highlightQuery = defineQuery([HighlightComponent, MeshComponent, GroupComponent, VisibleComponent])
+const highlightQuery = defineQuery([HighlightComponent, VisibleComponent])
 
 const execute = () => {
   /** @todo support multiple scenes */
   if (!hasComponent(Engine.instance.viewerEntity, RendererComponent)) return
 
-  // @ts-ignore @todo why does typescript freak out here?
-  getComponent(Engine.instance.viewerEntity, RendererComponent).effectComposer?.OutlineEffect?.selection.set(
-    highlightQuery().map((entity) => getComponent(entity, MeshComponent))
-  )
+  let highlightObjects = new Set<Object3D>()
+  for (const entity of highlightQuery()) {
+    traverseEntityNode(entity, (child, index) => {
+      if (!hasComponent(child, MeshComponent)) return
+      if (!hasComponent(child, GroupComponent)) return
```

Comment:
we need to exclude non visible entities here too

Reply:
implemented